### DISCUSSION TITLE
Query/path parameter validation not working with some Pydantic v2 types, e.g. PositveInt

### DISCUSSION THREAD

**Discussion by mblo** [fastapi/fastapi#10580] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import PositiveInt

app = FastAPI()

@app.get("/respond/{another_int}")
async def do_respond(input_int: PositiveInt = 1, another_int: PositiveInt = 1) -> int:
    return input_int + another_int + 1
```

### Description

Python 3.11.6
`pip install fastapi==0.104.1`

### Expected
Each of the following queries should respond with a 422 error
- `http://localhost:8000/respond/-1?input_int=2`
- `http://localhost:8000/respond/2?input_int=-1`
- `http://localhost:8000/respond/-1?input_int=-1`

### Working: FastApi + Pydantic v1
`pip install pydantic==1.10.13`
Works as expected - every example returns with a 422 error

### Not working: FastApi + Pydantic v2
`pip install pydantic==2.4.2`
Input validation does not work, returns http 200 with the dummy int calculation

I'm aware of the changed implementation of PositiveInt of Pydantic v1 vs v2. However, I don't see any reason why FastApi should not support the "new" PositiveInt implementation of Pydantic v2.
I have not checked all affected types, but e.g.  `NonNegativeInt` is also affected.

Related discussion: 
- input validation using `Form` https://github.com/tiangolo/fastapi/discussions/9607#discussioncomment-6149116

### Operating System

Linux

### Operating System Details

Python 3.11.6
Tested on `macOS 14` and `python:3.11-slim-bullseye`

### FastAPI Version

0.104.1

### Pydantic Version

2.4.2

### Python Version

3.11.6

**Comment by YuriiMotov** [comment]
This has been fixed and works well in current versions (FastAPI 0.115.12, Pydantic 2.11.5)
